### PR TITLE
Use a middleware to wrap the routes and catch the possible promise errors

### LINKED ISSUE #143: Switch to async handlers
Currently all api handlers are async and wrapped in try/catch, a better solution would be to create an async wrapper that returns unexpected errors as 500 or using something like https://github.com/spatools/express-async-router

Refs #135

### PR DESCRIPTION
**Pull request by gregory-latinier**
Fix #143

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -99,14 +99,16 @@ if (process.env.SENTRY_DSN) {
   app.use(Raven.errorHandler());
 }
 
-app.use((err, req, res) => {
+// eslint-disable-next-line no-unused-vars
+app.use((err, req, res, next) => {
+  console.log(err);
```
Comment (jnordberg) [inline app.js]: Use logger here with a descriptive error message

File: helpers/asyncMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const asyncMiddleware = fn =>
+  (req, res, next) => {
+    Promise.resolve(fn(req, res, next))
```
Comment (jnordberg) [inline helpers/asyncMiddleware.js]: Next should be called when promise resolves, not passed to the function

File: routes/api.js [hunk 3]
```diff
@@ -250,25 +232,21 @@ const rejectAccount = async (req, email) => {
...
+    type: 'create_account',
+    email,
+  }, process.env.JWT_SECRET, { expiresIn: '7d' });
+
+  req.mail.send(email, 'create_account', {
+    url: `${req.protocol}://${req.get('host')}/create-account?token=${mailToken}`,
+  },
+  (err) => {
+    if (err) {
+      throw new Error(err);
```
Comment (jnordberg) [inline routes/api.js]: This will be thrown outside of the handler

File: routes/api.js [hunk 4]
```diff
@@ -370,126 +342,106 @@ router.get('/guess_country', (req, res) => {
...
+          process.env.CREATE_ACCOUNT_DELEGATION,
+          process.env.DELEGATOR_USERNAME,
+          username,
+          owner,
+          active,
+          posting,
+          publicKeys.memo,
+          JSON.stringify({}),
+          [],
+          async (err) => {
```
Comment (jnordberg) [inline routes/api.js]: No callbacks allowed in async handlers

**Comment by jnordberg** [comment]
@gregory-latinier Is this something you're actively working on? If not I'll jump in and do it. It's a blocker for being able to debug the api endpoints.

**Comment by gregory-latinier** [comment]
I'm not on this no.
I'm stuck on the error thrown by the email. It is swallowed somewhere. I'm happy to let you find it !

**Comment by jnordberg** [comment]
👌 I'll have a stab at it

**Comment by jnordberg** [comment]
Superseded by #174

**Pull request closed without merging**